tests/wandb_artifacts_test.py

```diff
@@ -491,10 +493,10 @@ def test_deduplicate_wbimage_from_array(runner):
         artifact.add(wb_image_1, "my-image_1")
         artifact.add(wb_image_2, "my-image_2")
         artifact.add(wb_image_3, "my-image_3")
-        assert len(artifact.manifest.entries) == 5
+        assert len(artifact.manifest.entries) == 6
```

Comment:
This count (and the other count changed) are intentionally checking for deduplication of the same logical object. I think we might need to revert this change / change these counts back and re-add in the is_tmp check that was removed. Without this, we are going to be duplicating a bunch of data in artifacts (particularly the cases where images or other objects are modified at runtime). @annirudh tagging you since this was merged already.

Reply:
Thanks for calling this out, @tssweeney! Here's what was going through my mind when I was making the change:

- When adding a file by name (i.e.: `is_tmp` will be false), we still want to use the digest in some way to uniquely identify the file in the event we have two different paths with the same basename, like `dir1/img.png` and `dir2/img.png`.
- I thought we could do that in one place by moving the `is_tmp` logic up into the `data_types` layer.
- At the artifact level itself, we dedupe by digest so I (incorrectly) figured it was OK to have a duplicate entry, not realizing that this would mess up logging the same logical artifact again since the path would be randomly generated each time.

I fixed this and the tests in: https://github.com/wandb/client/pull/1661/files.

Reply:
Ah - makes sense - thanks for clarifying here. Also, for quickly putting up a new PR. I just read it over and looks like it properly enforces both requirements - sweet!